core/src/test_help/mod.rs

```diff
@@ -896,3 +897,34 @@ macro_rules! prost_dur {
             .expect("test duration fits")
     };
 }
+
+/// Initiate shutdown, drain the pollers, and wait for shutdown to complete.
+pub(crate) async fn drain_pollers_and_shutdown(worker: Worker) {
```

Comment:
Yeah.. I thought about it and just went with this because it was easier and I was in the phase of getting tests to pass.
I'll change, I like the extension trait.